test/suites/ecma6-tests/ecma6-destructuring-assignment-tests.js

```diff
@@ -0,0 +1,91 @@
+	});
+	
+	skip("Object destructuring: assignment with object declaration", function(assert) {
+	var result = puma.evalPuma("let o = {p: 42, q: true}; let {p, q} = o; p === 42 && q === true;");
+        assert.equal(result.value, true);
+	});
+	
+	skip("Object destructuring: assignment with object declaration", function(assert) {
+	var result = puma.evalPuma("let o = {p: 42, q: true}; let {p: foo, q: bar} = o; foo === 42 && bar === true;");
+        assert.equal(result.value, true);
+	});	
+	
+	skip("Object destructuring: assignment without declaration", function(assert) {
+        var result = puma.evalPuma("let a, b; ({a, b} = {a:1, b:2}); a === 1 && b === 2;");
+        assert.equal(result.value, true);
+	});
+	
+	skip("Object destructuring: function arguments default, simple example", function(assert) {
+        var result = puma.evalPuma("let {x = 1, y = 200} = {x: 100};");
+        assert.equal(result.value, true);
+	});	
+	
+	skip("Object destructuring: function arguments default", function(assert) {
+        var result = puma.evalPuma("function drawES2015Chart({size = 'big', cords = { x: 0, y: 0 }, radius = 25} = {}){return [size, cords, radius]} let [size, cords, radius] = drawES2015Chart({cords: { x: 18, y: 30 }, radius: 30}); size === 'big' && cords.x === 18 && cords.y === 30 && radius === 30;");
+        assert.equal(result.value, true);
+	});
+	
+	skip("Object destructuring: nested objects and array destructuring", function(assert) {
+        var result = puma.evalPuma("let metadata = {title: 'Scratchpad', translations: [{locale: 'de', localization_tags: [ ], last_edit: '2014-04-14T08:43:37', url: '/de/docs/Tools/Scratchpad', title: 'JavaScript-Umgebung'}], url: '/en-US/docs/Tools/Scratchpad'}; let { title: englishTitle, translations: [{ title: localeTitle }] } = metadata; englishTitle === 'Scratchpad' && localeTitle === 'JavaScript-Umgebung';");
+        assert.equal(result.value, true);
+	});
+	
+	skip("Object destructuring: For iterations with destructuring", function(assert) {
+        var result = puma.evalPuma("var people = [{name: 'Mike Smith',family: {mother: 'Jane Smith', father: 'Harry Smith'}, age: 35}, {name: 'Tom Jones', family: {mother: 'Norah Jones', father: 'Richard Jones'}, age: 25}]; let fathers=[]; for (let {name: n, family: { father: f } } of people) {father=[{name: n, father: f}];fathers.push(father);} fathers.length === 2;");
+        assert.equal(result.value, true);
+	});	
+	
+	skip("Object destructuring: Computed object property names and destructuring", function(assert) {
+        var result = puma.evalPuma("let key = 'z'; let { [key]: foo } = { z: 'bar' }; foo === 'bar';");
+        assert.equal(result.value, true);
+	});
+	
+	skip("Object destructuring: Computed object property names and destructuring, alternative", function(assert) {
+        var result = puma.evalPuma("let {['f' + 'oo']: there} = {foo : 'bar'}; there === 'bar';");
+        assert.equal(result.value, true);
+	});	
+	
```

Comment:
Indentation doesn't look right. ESLint isn't running on this folder, i'll be fixing that soon